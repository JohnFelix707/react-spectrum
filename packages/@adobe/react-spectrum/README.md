# @adobe/react-spectrum

This package allows you to easily import any component published under @react-spectrum. Think of it as a helpful alias,
with each component "pinned" to a specific version.

Components are updated sporadically, so if you require a change from a version of a component that is not yet included
in this package, you are able to install the specific version and then update your import statement to use the
component from the @react-spectrum scope. It's worth noting that this may result in increased file size.

## Naming conventions

Any component published under the @react-spectrum scope will be available at the same path from @adobe/react-spectrum.
For example, if you want to use `Checkbox` from @react-spectrum this is what you'd be importing:

```js
import {Checkbox} from '@react-spectrum/checkbox';
```

To import the same component from @adobe/react-spectrum, it would look like this:

```js
import {Checkbox} from '@adode/react-spectrum/checkbox';
```

This convention will be consistent for all packages to make it easy to switch between @react-spectrum and
@adobe/react-spectrum.

## Component versions

| Component                          | Version        |
| ---------------------------------- | -------------- |
| @react-spectrum/actiongroup        | 1.0.0          |
| @react-spectrum/breadcrumbs        | 1.0.0          |
| @react-spectrum/button             | 1.0.0          |
| @react-spectrum/buttongroup        | 1.0.0          |
| @react-spectrum/checkbox           | 1.0.0          |
| @react-spectrum/dialog             | 1.0.0          |
| @react-spectrum/divider            | 1.0.0          |
| @react-spectrum/form               | 1.0.0          |
| @react-spectrum/icon               | 1.0.0          |
| @react-spectrum/illustratedmessage | 1.0.0          |
| @react-spectrum/image              | 1.0.0          |
| @react-spectrum/label              | 1.0.0          |
| @react-spectrum/layout             | 1.0.0          |
| @react-spectrum/link               | 1.0.0          |
| @react-spectrum/listbox            | 1.0.0          |
| @react-spectrum/menu               | 1.0.0          |
| @react-spectrum/meter              | 1.0.0          |
| @react-spectrum/overlays           | 1.0.0          |
| @react-spectrum/picker             | 1.0.0          |
| @react-spectrum/progress           | 1.0.0          |
| @react-spectrum/provider           | 1.0.0          |
| @react-spectrum/radio              | 1.0.0          |
| @react-spectrum/searchfield        | 1.0.0          |
| @react-spectrum/statuslight        | 1.0.0          |
| @react-spectrum/switch             | 1.0.0          |
| @react-spectrum/textfield          | 1.0.0          |
| @react-spectrum/theme-default      | 1.0.0          |
| @react-spectrum/utils              | 1.0.0          |
| @react-spectrum/view               | 1.0.0          |
| @react-spectrum/well               | 1.0.0          |
