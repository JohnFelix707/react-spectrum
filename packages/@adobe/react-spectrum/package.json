{
  "name": "@adobe/react-spectrum",
  "version": "3.38.0",
  "description": "Spectrum UI components in React",
  "license": "Apache-2.0",
  "main": "dist/main.js",
  "module": "dist/module.js",
  "exports": {
    ".": {
      "types": "./dist/types.d.ts",
      "import": "./dist/import.mjs",
      "require": "./dist/main.js"
    },
    "./i18n": {
      "types": "./i18n/index.d.ts",
      "import": "./i18n/index.mjs",
      "require": "./i18n/index.js"
    },
    "./i18n/*": {
      "types": "./i18n/lang.d.ts",
      "import": "./i18n/*.mjs",
      "require": "./i18n/*.js"
    }
  },
  "types": "dist/types.d.ts",
  "source": "src/index.ts",
  "files": [
    "dist",
    "i18n"
  ],
  "sideEffects": false,
  "scripts": {
    "prepublishOnly": "mkdir -p dist; cp src/index.ts dist/types.d.ts; grep -v '^export type' src/index.ts > dist/module.js; babel --root-mode upward src/index.ts -o dist/main.js"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/adobe/react-spectrum"
  },
  "dependencies": {
    "@internationalized/string": "^3.2.5",
    "@react-aria/i18n": "^3.12.4",
    "@react-aria/ssr": "^3.9.7",
    "@react-aria/utils": "^3.26.0",
    "@react-aria/visually-hidden": "^3.8.18",
    "@react-spectrum/accordion": "^3.0.0",
    "@react-spectrum/actionbar": "^3.6.2",
    "@react-spectrum/actiongroup": "^3.10.10",
    "@react-spectrum/avatar": "^3.0.17",
    "@react-spectrum/badge": "^3.1.18",
    "@react-spectrum/breadcrumbs": "^3.9.12",
    "@react-spectrum/button": "^3.16.9",
    "@react-spectrum/buttongroup": "^3.6.17",
    "@react-spectrum/calendar": "^3.5.0",
    "@react-spectrum/checkbox": "^3.9.11",
    "@react-spectrum/color": "^3.0.2",
    "@react-spectrum/combobox": "^3.14.0",
    "@react-spectrum/contextualhelp": "^3.6.16",
    "@react-spectrum/datepicker": "^3.11.0",
    "@react-spectrum/dialog": "^3.8.16",
    "@react-spectrum/divider": "^3.5.18",
    "@react-spectrum/dnd": "^3.5.0",
    "@react-spectrum/dropzone": "^3.0.6",
    "@react-spectrum/filetrigger": "^3.0.6",
    "@react-spectrum/form": "^3.7.10",
    "@react-spectrum/icon": "^3.8.0",
    "@react-spectrum/illustratedmessage": "^3.5.5",
    "@react-spectrum/image": "^3.5.6",
    "@react-spectrum/inlinealert": "^3.2.10",
    "@react-spectrum/labeledvalue": "^3.1.18",
    "@react-spectrum/layout": "^3.6.10",
    "@react-spectrum/link": "^3.6.12",
    "@react-spectrum/list": "^3.9.0",
    "@react-spectrum/listbox": "^3.14.0",
    "@react-spectrum/menu": "^3.21.0",
    "@react-spectrum/meter": "^3.5.5",
    "@react-spectrum/numberfield": "^3.9.8",
    "@react-spectrum/overlays": "^5.7.0",
    "@react-spectrum/picker": "^3.15.4",
    "@react-spectrum/progress": "^3.7.11",
    "@react-spectrum/provider": "^3.10.0",
    "@react-spectrum/radio": "^3.7.11",
    "@react-spectrum/searchfield": "^3.8.11",
    "@react-spectrum/slider": "^3.7.0",
    "@react-spectrum/statuslight": "^3.5.17",
    "@react-spectrum/switch": "^3.5.10",
    "@react-spectrum/table": "^3.15.0",
    "@react-spectrum/tabs": "^3.8.15",
    "@react-spectrum/tag": "^3.2.11",
    "@react-spectrum/text": "^3.5.10",
    "@react-spectrum/textfield": "^3.12.7",
    "@react-spectrum/theme-dark": "^3.5.14",
    "@react-spectrum/theme-default": "^3.5.14",
    "@react-spectrum/theme-light": "^3.4.14",
    "@react-spectrum/tooltip": "^3.7.0",
    "@react-spectrum/view": "^3.6.14",
    "@react-spectrum/well": "^3.4.18",
    "@react-stately/collections": "^3.12.0",
    "@react-stately/data": "^3.12.0",
    "@react-types/shared": "^3.26.0",
    "client-only": "^0.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0-rc.1 || ^18.0.0 || ^19.0.0-rc.1",
    "react-dom": "^16.8.0 || ^17.0.0-rc.1 || ^18.0.0 || ^19.0.0-rc.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.12.10",
    "@babel/core": "^7.24.3"
  }
}
