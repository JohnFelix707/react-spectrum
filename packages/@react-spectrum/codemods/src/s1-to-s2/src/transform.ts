import {CodemodOptions} from '../..';
import {run as jscodeshift} from 'jscodeshift/src/Runner.js';
import path from 'node:path';

const transformPath = path.join(__dirname, 'codemods', 'codemod.js');


export async function transform(options: CodemodOptions) {
  let {
    path: filePath = '.',
    ...rest
  } = options;

  return await jscodeshift(transformPath, [filePath], rest);
}
