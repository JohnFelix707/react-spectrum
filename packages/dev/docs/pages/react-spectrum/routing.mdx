{/* Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {Layout} from '@react-spectrum/docs';
export default Layout;

import {Keyboard} from '@react-spectrum/text';

---
category: Concepts
keywords: [router, links, frameworks, next.js, remix, react-router]
order: 5
---

# Client Side Routing

Many React Spectrum components support rendering as HTML links. This page discusses how to set up your app to integrate React Spectrum links with your framework or client side router.

## Introduction

React Spectrum components such as [Menu](Menu.html), [Tabs](Tabs.html), [TableView](TableView.html), and many others support rendering elements as links that perform navigation when the user interacts with them. Each component that supports link behavior accepts the `href` prop, which causes the component to render an `<a>` element. Other link DOM props such as `target` and `download` are also supported.

By default, links perform native browser navigation when they are interacted with. However, many apps and frameworks use client side routers to avoid a full page reload when navigating between pages. The `RouterProvider` component configures all React Spectrum components within it to navigate using the client side router you provide. Set this up once in the root of your app, and any React Spectrum component with the `href` prop will automatically navigate using your router.

Note that external links to different origins will not trigger client side routing, and will use native browser navigation. Additionally, if the link has a [target](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target) other than `"_self"`, uses the [download](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#download) attribute, or the user presses modifier keys such as <Keyboard>Command</Keyboard> or <Keyboard>Alt</Keyboard> to change the default behavior, browser native navigation will occur instead of client side routing.

## RouterProvider

The `RouterProvider` component accepts a single prop: `navigate`. This should be set to a function received from your router for performing a client side navigation programmatically. The following example shows the general pattern. Framework-specific examples are shown below.

```tsx
import {RouterProvider} from '@adobe/react-spectrum';

function App() {
  let navigate = useNavigateFromYourRouter();

  return (
    <RouterProvider navigate={navigate}>
      {/* ... */}
    </RouterProvider>
  );
}
```

### React Router

The [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook from `react-router-dom` returns a `navigate` function you can pass to `RouterProvider`. Ensure that the component that calls `useNavigate` and renders `RouterProvider` is inside the router component (e.g. `BrowserRouter`) so that it has access to React Router's internal context. The React Router `<Routes>` element should also be defined inside React Spectrum's `<RouterProvider>` so that links inside the rendered routes have access to the router.

```tsx
import {BrowserRouter, useNavigate} from 'react-router-dom';
import {RouterProvider} from '@adobe/react-spectrum';

function App() {
  let navigate = useNavigate();

  return (
    <RouterProvider navigate={navigate}>
      {/* Your app here... */}
      <Routes>
        <Route path="/" element={<HomePage />} />
        {/* ... */}
      </Routes>
    </RouterProvider>
  );
}

<BrowserRouter>
  <App />
</BrowserRouter>
```

### Next.js

#### App router

The [useRouter](https://nextjs.org/docs/app/api-reference/functions/use-router) hook from `next/navigation` returns a router object that can be used to perform navigation. `RouterProvider` should be rendered from a client component at the root of each page or layout that includes React Spectrum links. You can create a new client component for this, or combine it with other top-level providers as described in the [Next.js docs](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#using-context-providers).

```tsx
// app/provider.tsx
"use client";

import {useRouter} from 'next/navigation';
import {RouterProvider} from '@adobe/react-spectrum';

export function ClientProviders({children}) {
  let router = useRouter();

  return (
    <RouterProvider navigate={router.push}>
      {children}
    </RouterProvider>
  );
}
```

Then, in your page or layout server component, wrap your app in the `ClientProviders` component that you defined.

```tsx
// app/layout.tsx
import {AppProviders} from './provider';

export default function RootLayout({children}) {
  return (
    <html>
      <body>
        <ClientProviders>{children}</ClientProviders>
      </body>
    </html>
  );
}
```

#### Pages router

The [useRouter](https://nextjs.org/docs/pages/api-reference/functions/use-router) hook from `next/router` returns a router object that can be used to perform navigation. `RouterProvider` should be rendered at the root of each page that includes React Spectrum links, or in `pages/_app.tsx` to add it to all pages.

```tsx
// pages/_app.tsx
import type { AppProps } from 'next/app';
import {useRouter} from 'next/router';
import {RouterProvider} from '@adobe/react-spectrum';

export default function MyApp({Component, pageProps}: AppProps) {
  let router = useRouter();

  return (
    <RouterProvider navigate={router.push}>
      <Component {...pageProps} />
    </RouterProvider>
  );
}
```

### Remix

Remix uses React Router under the hood, so the same [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook described above also works in Remix apps. `RouterProvider` should be rendered at the root of each page that includes React Spectrum links, or in `app/root.tsx` to add it to all pages. See the [Remix docs](https://remix.run/docs/en/main/file-conventions/root) for more details.

```tsx
// app/root.tsx
import {useNavigate, Outlet} from '@remix-run/react';
import {RouterProvider} from '@adobe/react-spectrum';

export default function App() {
  let navigate = useNavigate();

  return (
    <html lang="en">
      <head>
        {/* ... */}
      </head>
      <body>
        <RouterProvider navigate={navigate}>
          <Outlet />
        </RouterProvider>
        {/* ... */}
      </body>
    </html>
  );
}
```
