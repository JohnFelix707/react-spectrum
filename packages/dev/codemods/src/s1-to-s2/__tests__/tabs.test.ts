// @ts-ignore
import {defineSnapshotTest} from 'jscodeshift/dist/testUtils';
import transform from '../src/codemods/codemod';

const test = (name: string, input: string) => {
  defineSnapshotTest(transform, {}, input, name);
};

test('Update to use new API', `
import {Tabs, TabList, TabPanels, Item} from '@adobe/react-spectrum';

<Tabs aria-label="History of Ancient Rome">
  <TabList>
    <Item key="FoR">Founding of Rome</Item>
    <Item key="MaR">Monarchy and Republic</Item>
    <Item key="Emp">Empire</Item>
  </TabList>
  <TabPanels>
    <Item key="FoR">
      Arma virumque cano, Troiae qui primus ab oris.
    </Item>
    <Item key="MaR">
      Senatus Populusque Romanus.
    </Item>
    <Item key="Emp">
      Alea jacta est.
    </Item>
  </TabPanels>
</Tabs>
`);

test('Remove isEmphasized', `
import {Tabs, TabList, TabPanels, Item} from '@adobe/react-spectrum';
let isEmphasized = true;
let props = {isEmphasized: true};
<div>
  <Tabs aria-label="History of Ancient Rome" isEmphasized>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" isEmphasized={true}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" isEmphasized={false}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" isEmphasized={isEmphasized}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" {...props}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
</div>
`);

test('Remove isQuiet', `
import {Tabs, TabList, TabPanels, Item} from '@adobe/react-spectrum';
let isQuiet = true;
let props = {isQuiet: true};
<div>
  <Tabs aria-label="History of Ancient Rome" isQuiet>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" isQuiet={true}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" isQuiet={false}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" isQuiet={isQuiet}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
  <Tabs aria-label="History of Ancient Rome" {...props}>
    <TabList>
      <Item key="FoR">Founding of Rome</Item>
      <Item key="MaR">Monarchy and Republic</Item>
      <Item key="Emp">Empire</Item>
    </TabList>
    <TabPanels>
      <Item key="FoR">
        Arma virumque cano, Troiae qui primus ab oris.
      </Item>
      <Item key="MaR">
        Senatus Populusque Romanus.
      </Item>
      <Item key="Emp">
        Alea jacta est.
      </Item>
    </TabPanels>
  </Tabs>
</div>
`);
