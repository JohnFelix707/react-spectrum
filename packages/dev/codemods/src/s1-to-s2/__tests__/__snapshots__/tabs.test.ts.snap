// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove isEmphasized 1`] = `
"import { Tab, TabPanel, Tabs, TabList } from "@react-spectrum/s2";
import { TabPanels, Item } from '@adobe/react-spectrum';
let isEmphasized = true;
let props = {isEmphasized: true};
<div>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome" // TODO(S2-upgrade): check this spread for style props
  {...props}><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
</div>"
`;

exports[`Remove isQuiet 1`] = `
"import { Tab, TabPanel, Tabs, TabList } from "@react-spectrum/s2";
import { TabPanels, Item } from '@adobe/react-spectrum';
let isQuiet = true;
let props = {isQuiet: true};
<div>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome"><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
  <Tabs aria-label="History of Ancient Rome" // TODO(S2-upgrade): check this spread for style props
  {...props}><TabList>
      <Tab id="FoR">Founding of Rome</Tab>
      <Tab id="MaR">Monarchy and Republic</Tab>
      <Tab id="Emp">Empire</Tab>
    </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>
</div>"
`;

exports[`Update to use new API 1`] = `
"import { Tab, TabPanel, Tabs, TabList } from "@react-spectrum/s2";
import { TabPanels, Item } from '@adobe/react-spectrum';

<Tabs aria-label="History of Ancient Rome"><TabList>
    <Tab id="FoR">Founding of Rome</Tab>
    <Tab id="MaR">Monarchy and Republic</Tab>
    <Tab id="Emp">Empire</Tab>
  </TabList><TabPanel id="FoR">Arma virumque cano, Troiae qui primus ab oris.</TabPanel><TabPanel id="MaR">Senatus Populusque Romanus.</TabPanel><TabPanel id="Emp">Alea jacta est.</TabPanel></Tabs>"
`;
